src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php

```diff
@@ -93,13 +93,18 @@ private function generateDeclaredRoutes()
             $properties[] = $compiledRoute->getHostTokens();
             $properties[] = $route->getSchemes();
 
-            $routes .= sprintf("        '%s' => %s,\n", $name, str_replace("\n", '', var_export($properties, true)));
+            $routes .= sprintf("                '%s' => %s,\n", $name, $this->exportArray($properties));
         }
-        $routes .= '    )';
+        $routes .= '            )';
 
         return $routes;
     }
 
+    protected function exportArray($array)
+    {
+        return preg_replace(array('/array \(/', '/,\n *\)/', '/,\n */', '/\n */'), array('array(', ')', ', ', ''), var_export($array, true));
```

Comment:
You are right. I thought I had this covered, but I must have missed something (I wrote this code a while back, so I don't remember my reasoning).

I cannot think of an easy way to fix this, so I think I'll just close this PR.

BTW the existing code simply strips newlines – this is also not completely kosher, but it might be enough for the specific arrays in question.